### PR TITLE
Remove WP.com themes list for not capable Jetpack plugins.

### PR DESCRIPTION
**Pull request by budzanowski**
### Info
We can only offer WordPress.com themes installation to users with Jetpack plugin >= 4.4.2 For earlier versions we need to remove WP.com themes lists and do not filter wpcom themes from the first list.

### Testing
You need two Jetpack sites one with version lower than 4.4.2 and one with that one or newer. 
If you don't have Jetpack with proper version you can always change required version in
` hasJetpackSiteJetpackThemesExtendedFeatures`  for testing purposes.

Open `/design` for Jetpack site:
- for >= 4.4.2 two lists of themes will be visible, first list will have wpcom themes filtered
- for < 4.4.2 one list of themes will be visible and it will have all themes installed on that Jetpack site visible

### CODE REVIEW

File: client/state/themes/utils.js [hunk 1]
```diff
@@ -204,31 +204,34 @@ export function isWpcomTheme( themeId ) {
...
- * @param  {Array}  themes Array of theme objects
- * @param  {Object} query  Themes query
- * @return {Array}         Filtered themes
+ * @param  {Array}   themes      Array of theme objects
+ * @param  {Object}  query       Themes query
+ * @param  {Boolean} filterWpcom Switch to filter out themes with -wpcom suffix
+ * @return {Array}               Filtered themes
  */
-export function filterThemesForJetpack( themes, query ) {
+export function filterThemesForJetpack( themes, query, filterWpcom = false ) {
```
Comment (ockham) [inline client/state/themes/utils.js]: Hrm. I'm not too happy about putting two different kinds of filter into the same function and switch one of them on and off with a flag. I'd rather separate concerns, i.e. have a separate filter function for each, and use them subsequently.

Comment (ockham) [inline client/state/themes/utils.js]: Also, can't we make them both just predicates and just inline the `filter` call? I'd rather avoid having a filter function that really just combines lodash's `filter` (i.e. a general-purpose utility) with a predicate -- instead, I'd prefer making that explicit.

Comment (budzanowski) [inline client/state/themes/utils.js]: We already have them as predicates. So this should be easy.

**Comment by seear** [comment]
Since we only show one list for jetpack sites in production at the moment, why does this PR need to do anything more than add a jetpack version check alongside every current check for `config.isEnabled( 'manage/themes/upload' )` ?

**Comment by budzanowski** [comment]
It is basically is just that( adding jetpack version check alongside every current check for `config.isEnabled( 'manage/themes/upload' )` with just a splash of refactoring.

**Comment by seear** [comment]
This PR seems to fix a problem I had noticed with the filter where activating a wpcom theme on a jetpack site would add the theme to the top list until a refresh. Nice.

**Comment by budzanowski** [comment]
Rebased.

**Comment by ockham** [comment]
Okay, it took me a while to think it thru, but I _think_ it'd be cool if we kept the `hasJetpackSiteJetpackThemesExtendedFeatures` check in one place. Since it has a direct UI implication, it's good to have it in `themes-selection`. Since we filter from the top list as a consequence of that UI implication (the second list being there), I think that filtering should also be done there -- in `ThemesSelection`'s `connect` statement, that is. (The filtering of JP themes for the given query should stay in the selector -- it is making up for a shortcoming of the endpoint, so it should remain "close" to the endpoint, in terms of data layers.) This would have the nice side effect that at least we wouldn't need to use `getState` in `requestThemes`.

I realize however that I've taken too long to properly review this PR, so if the general feeling is to rather get this finally in, I'm okay with 🚢 ping it so we're at least covering the functionality; we can defer the polishing to a later PR. (I think that's actually what I'm suggesting -- let's ship this, and move the filtering around in a subsequent PR.)

**Comment by ockham** [comment]
> Since we filter from the top list as a consequence of that UI implication (the second list being there), I think that filtering should also be done there -- in `ThemesSelection`'s `connect` statement, that is. 

Hmm, then again, handling of things like pagination probably becomes harder if we filter at `connect` level instead of requesting/receiving action level... I might have to retract my suggestion...

**Comment by ockham** [comment]
Also, a lot of these things are workarounds for missing pagination and query capabilities on the Jetpack side of things. So maybe we should really add those at some point...

**Pull request merged**